File: recurrent_events/preprocessEv.m
function [lambda_ti, survival]= preprocessEv(tev, T,w)
lambda_ti=zeros(1,length(tev));
survival=0;
for ii=1:length(tev)
    if ii>1
        lambda_ti(ii)= sum(exp(-w*(tev(ii)-tev(1:ii-1))));
    end
    survival=survival+ (1/w)*(1-exp(-w*(T-tev(ii))));
end

File: recurrent_events/preprocessEv.py
import numpy as np

def preprocessEv(tev, T, w):
    lambda_ti = np.zeros_like(tev, dtype=float)
    survival = 0

    for i in range(len(tev)):
        lambda_ti[i] = np.sum(np.exp(-w * (tev[i] - tev[0:i])))
        survival += (1.0 / w) * (1.0 - np.exp(-w * (T - tev[i])))

    return lambda_ti, survival

